src/app/space/create/deployments/resource-usage/resource-card.component.ts

```diff
@@ -1,44 +1,59 @@
 import {
   Component,
   Input,
+  OnDestroy,
   OnInit
 } from '@angular/core';
 
-import { Observable } from 'rxjs';
+import { Observable, Subscription } from 'rxjs';
 
-import { MemoryStat } from 'app/space/create/deployments/models/memory-stat';
 import { Environment } from '../models/environment';
+import { MemoryStat } from '../models/memory-stat';
+import { Stat } from '../models/stat';
 import { DeploymentsService } from '../services/deployments.service';
 
 @Component({
   selector: 'resource-card',
   templateUrl: 'resource-card.component.html'
 })
-export class ResourceCardComponent implements OnInit {
+export class ResourceCardComponent implements OnDestroy, OnInit {
 
-  @Input() spaceId: string;
+  @Input() spaceId: Observable<string>;
   @Input() environment: Environment;
 
   memUnit: Observable<string>;
   active: boolean = false;
 
+  cpuStat: Observable<Stat>;
+  memStat: Observable<Stat>;
+
+  private subscriptions: Subscription[] = [];
+
   constructor(
     private deploymentsService: DeploymentsService
   ) { }
 
   ngOnInit(): void {
-    this.deploymentsService
-    .isDeployedInEnvironment(this.spaceId, this.environment.name)
-    .subscribe((active: boolean) => {
-      this.active = active;
-      if (active) {
-        this.memUnit = this.deploymentsService.getEnvironmentMemoryStat(this.spaceId, this.environment.name)
-        .first()
-        .map((stat: MemoryStat) => stat.units);
-      }
-    });
-
+    this.subscriptions.push(this.spaceId.subscribe((spaceId: string) => {
+      this.subscriptions.push(this.deploymentsService
+      .isDeployedInEnvironment(spaceId, this.environment.name)
+      .subscribe((active: boolean) => {
+        this.active = active;
+        if (active) {
+          this.memUnit = this.deploymentsService.getEnvironmentMemoryStat(spaceId, this.environment.name)
+          .first()
+          .map((stat: MemoryStat) => stat.units);
+        }
+      }));
+      this.cpuStat = this.deploymentsService.getEnvironmentCpuStat(spaceId, this.environment.name);
+      this.memStat = this.deploymentsService.getEnvironmentMemoryStat(spaceId, this.environment.name);
```

Comment:
Assigning Observable properties only after some other Observable has emitted a value is what got us into trouble the first time around when the spaceId schedular changed in Dana's `modify-rxjs-behavior` branch. If we can't work under the assumption that these things resolve synchronously on subscribe, then it's possible that we will be instantiating child components with not-yet-defined bindings for Observables (like `memStat` here).

IMO the clearest solution to this problem is to ensure that all component properties are defined synchronously by the end of `ngOnInit` at the latest. For the case of `memStat` for example, this would mean immediately defining it as a `Subject`, which would then be passed to the component's children, and they can subscribe immediately even if the spaceId has not yet resolved. This component would then create its own subscription via `deploymentsService.getEnvironmentMemoryStat(spaceId, this.environment.name).subscribe(this.memorySubject)`, so that service emissions are relayed to child components.

Reply:
Good point. I'll rework this and use Dana's change to test it. I'll ping you again when it's ready for review